### ISSUE TITLE
Alter table rename column does not rename index column

### ISSUE THREAD

**Issue by robertpang** [yugabyte/yugabyte-db#506]
Jira Link: [DB-664](https://yugabyte.atlassian.net/browse/DB-664)
When an index is created on a table and then the column indexed is renamed, the column name in the index is not renamed:

```
cqlsh:k> create table t (k int primary key, v int) with transactions = { 'enabled' : true };
cqlsh:k> create index i on t (v);
cqlsh:k> describe t;

CREATE TABLE k.t (
    k int PRIMARY KEY,
    v int
) WITH default_time_to_live = 0
    AND transactions = {'enabled': 'true'};
CREATE INDEX i ON k.t (v, k)
    WITH transactions = {'enabled': 'true'};

cqlsh:k> alter table t rename v to w;
cqlsh:k> describe t;

CREATE TABLE k.t (
    k int PRIMARY KEY,
    w int
) WITH default_time_to_live = 0
    AND transactions = {'enabled': 'true'};
CREATE INDEX i ON k.t (w, k)
    WITH transactions = {'enabled': 'true'};

cqlsh:k> select * from system_schema.columns where keyspace_name = 'k';

 keyspace_name | table_name | column_name | clustering_order | column_name_bytes | kind          | position | type
---------------+------------+-------------+------------------+-------------------+---------------+----------+------
             k |          t |           k |             none |              null | partition_key |        0 |  int
             k |          t |           w |             none |              null |       regular |       -1 |  int
             k |          i |           v |             none |              null | partition_key |        0 |  int
             k |          i |           k |              asc |              null |    clustering |        0 |  int
```

**Comment by jaki** [comment]
Ran this recently and got slightly different output on describe:

```
ycqlsh> create keyspace k; use k;
ycqlsh:k> create table t (k int primary key, v int) with transactions = { 'enabled' : true };
ycqlsh:k> create index i on t (v);
ycqlsh:k> desc t;

CREATE TABLE k.t (
    k int PRIMARY KEY,
    v int
) WITH default_time_to_live = 0
    AND transactions = {'enabled': 'true'};
CREATE INDEX i ON k.t (v, k)
    WITH transactions = {'enabled': 'true'};

ycqlsh:k> alter table t rename v to w;
ycqlsh:k> desc t;

CREATE TABLE k.t (
    k int PRIMARY KEY,
    w int
) WITH default_time_to_live = 0
    AND transactions = {'enabled': 'true'};
CREATE INDEX i ON k.t (v, k)
    WITH transactions = {'enabled': 'true'};

ycqlsh:k> select * from system_schema.columns where keyspace_name = 'k';

 keyspace_name | table_name | column_name | clustering_order | column_name_bytes | kind          | position | type
---------------+------------+-------------+------------------+-------------------+---------------+----------+------
             k |          i |           v |             none |              null | partition_key |        0 |  int
             k |          i |           k |              asc |              null |    clustering |        0 |  int
             k |          t |           k |             none |              null | partition_key |        0 |  int
             k |          t |           w |             none |              null |       regular |       -1 |  int
```

Notice that the CREATE INDEX statement in describe shows v rather than w.

**Comment by jaki** [comment]
I tried a similar test for YSQL, and the difference is that the PG metadata for the index seems to have the correct attname (try `SELECT attname FROM pg_attribute INNER JOIN pg_class ON pg_class.oid = pg_attribute.attrelid WHERE relname = 't'` and `\d t`).

However, in both YCQL and YSQL, the docdb schemas of the indexes use the old column name.

**Comment by jaki** [comment]
If we do decide to fix the docdb schemas to have the new column names, then make sure that client's alter waits on indexes as well (issue #8774, #8869).

**Comment by tverona1** [comment]
Tagging this as YCQL. YSQL behavior (from postgres perspective) looks correct:
```
yugabyte=# create table t (k int primary key, v int);
CREATE TABLE
yugabyte=# create index i on t (v);
CREATE INDEX
yugabyte=# \d+ t

                                     Table "public.t"
 Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description 
--------+---------+-----------+----------+---------+---------+--------------+-------------
 k      | integer |           | not null |         | plain   |              | 
 v      | integer |           |          |         | plain   |              | 
Indexes:
    "t_pkey" PRIMARY KEY, lsm (k HASH)
    "i" lsm (v HASH)

yugabyte=# alter table t rename v to w;
ALTER TABLE
yugabyte=# \d+ t
                                     Table "public.t"
 Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description 
--------+---------+-----------+----------+---------+---------+--------------+-------------
 k      | integer |           | not null |         | plain   |              | 
 w      | integer |           |          |         | plain   |              | 
Indexes:
    "t_pkey" PRIMARY KEY, lsm (k HASH)
    "i" lsm (w HASH)

```

**Comment by dr0pdb** [comment]
Per the discussion on slack - the immediate fix will be to disallow all ALTER column options that has an INDEX with a GFlag to enable it if the user relies on the existing behavior.

**Comment by dr0pdb** [comment]
Opened a separate issue for the long term fix: #16804
